core/src/main/java/org/elasticsearch/common/network/NetworkAddress.java

```diff
@@ -180,4 +182,13 @@ static String format(InetAddress address, int port, boolean includeHost) {
         
         return builder.toString();
     }
+    
+    @SuppressForbidden(reason = "we call toString to avoid a DNS lookup")
+    private static String getInet6HostAddress(InetAddress address) {
+        return address.getHostAddress().replaceAll(":0+([0-9])",":$1")
+                .replaceAll("((?::0\\b){2,}):?(?!\\S*\\b\\1:0\\b)(\\S*)", "::$2")
+                .replaceFirst("^0::", "::")
+                .replaceFirst("%\\d+", "");
+        // see http://stackoverflow.com/a/7044170/4746692
```

Comment:
Yeah, I agree with the issue with regex formatting.

Regarding incorrect formatting, the tests did look like they covered all the scenarios.